drafts/two-way-binding-signals-and-query-params.md

```diff
@@ -0,0 +1,754 @@
+
+Effects are executed as soon as they are defined, this is necessary so they can keep track of the signals that are used inside of them.
+
+When the second effect is initially executed, a navigation will be triggered, cancelling Angular's start up navigation and losing with it the actual values of the query params. 
+
+Let's try to understand it better with a specific example: Let's say we are visiting `http://localhost:4200?firstName=Dave&lastName=Doe&age=40`. And our `AppComponent` has the following code:
+
+```ts
+firstName = this.queryParamService.get("firstName");
+```
+
+Notice that it's only creating a single signal. In this case to track the `firstName` query param, but we will still experience the empty UI situation, affecting all query params in general.
+
+When `queryParamService.get("firstName")` is executed we create a writable `queryParamValue` signal linked to the `firstName` query param. During this process we define two effects and they get executed immediately to keep track of their corresponding signals as we have mentioned before. 
+
+Notice how we trigger a `router.navigate` in the second effect:
+
+```ts
+effect(() =>
+  this.router.navigate([], {
+    queryParams: { [queryParamName]: queryParamValue() },
+    queryParamsHandling: "merge",
+  })
+);
+```
+
+This navigation will be triggered right away. And since our `AppComponent` is rendered on start up, it will definitely conflict with Angular's initial navigation. 
+
+Therefore, Angular will cancel the current navigation. In this case, the initial one, the one that had this information: "navigate to `http://localhost:4200?firstName=Dave&lastName=Doe&age=40`" and will be replaced with "navigate to `http://localhost:4200/`", with no query params information at all. 
+
+As soon as this navigation finishes Angular could emit the actual query param values through the `activatedRoute.queryParams` observable, but there are no query params in `http://localhost:4200/`.
+
+
+The reason why there are no query params in this new navigation is in the `router.navigate` we are triggering in the effect:
+
+```ts
+  queryParams: { [queryParamName]: queryParamValue() },
+```
+
+In this example `queryParamName` is `"firstName"` and `queryParamValue()` is initially `undefined` for what we mentioned a bit above (you could think that we are accessing `activatedRoute.snapshot.queryParams["firstName"]` during the period of time in which the query params object is empty `{}`). 
+
+So the effect is triggering a navigation that has this information: "navigate to `http://localhost:4200?firstName=undefined`", since the value of `firstName` is `undefined` it's just removed from the query params. And notice that there's no information about the other query params at all.
+
+We could argue that we are indicating Angular to `merge` the query params in that `router.navigate` call. So no query params should be removed from the browser url.
```

Comment:
**thought:** I think the whole bit starting with "Let's try to understand …" on line 471 is a bit overkill, it's overly detailed and repetitive. Readers that came for Signals will jump off here. Try to reduce it to the absolute minimum explaining the problem but give readers some credit :D